test/infrastructure/docker/internal/controllers/backends/docker/dockermachine_backend.go

```diff
@@ -179,32 +204,28 @@ func (r *MachineBackendReconciler) ReconcileNormal(ctx context.Context, cluster
 		dockerMachine.Status.Backend.Docker.LoadBalancerConfigured = true
 	}
 
-	// Update the ContainerProvisionedCondition condition
-	// NOTE: it is required to create the patch helper before this change otherwise it wont surface if
-	// we issue a patch down in the code (because if we create patch helper after this point the ContainerProvisionedCondition=True exists both on before and after).
-	newPatchHelperFunc := r.NewPatchHelperFunc
-	if newPatchHelperFunc == nil {
-		newPatchHelperFunc = patch.NewHelper
+	// Update the ContainerProvisioned and BootstrapExecSucceeded condition if not already in the correct state.
+	requeue := false
+	if conditions.GetReason(dockerMachine, infrav1.ContainerProvisionedCondition) != "" {
```

Comment:
I wonder if it would be better to check for status false here instead of reason != ""

(that's where I actually wanted to put this comment :))

Reply:
Just feels less brittle to say. If this conditions is false, let's set it to true.

Compared to: Let's assume if the reason is not empty, then the condition is false, so let's set it to true

(If I'm not missing anything)